### PR TITLE
Remove OwnerReferences from CA configmaps

### PR DESCRIPTION
**Pull request by rubenvp8510**
Signed-off-by: Ruben Vargas <ruben.vp8510@gmail.com>

This change is to avoid `ConfigMaps` which are not in the same namespace as the operator from being garbage collected.

As the documentation says:

> 
> Cross-namespace owner references are disallowed by design.
> Namespaced dependents can specify cluster-scoped or namespaced owners. A namespaced owner must exist in the same namespace as the dependent. If it does not, the owner reference is treated as absent, and the dependent is subject to deletion once all owners are verified absent.

https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/

## Which problem is this PR solving?
- This solves an issue that deletes CA/Service CA `ConfigMaps` from sidecars that are not in the same namespace as the operator.

## Short description of the changes
- Removed the `OwnerReferences` from `ConfigMaps`
- Implemented a deletion logic for those `ConfigMaps` when a `Jaeger` instance is deleted.

**NOTE**: The clean up process consider that there is no more than one Jaeger instance with the same name on the cluster. I think we did this assumption for other parts of the code, so I don't think this will be a problem for now.

### CODE REVIEW

**Comment by rubenvp8510** [comment]
Struggling to reach the desired coverage due the tests can't reach error blocks.

File: pkg/controller/jaeger/configmap.go [hunk 1]
```diff
@@ -3,14 +3,18 @@ package jaeger
 import (
 	"context"
 
+	"k8s.io/apimachinery/pkg/selection"
```
Comment (jpkrohling) [inline pkg/controller/jaeger/configmap.go:6]: Import order.

File: pkg/controller/jaeger/configmap.go [hunk 2]
```diff
@@ -63,3 +67,30 @@ func (r *ReconcileJaeger) applyConfigMaps(ctx context.Context, jaeger v1.Jaeger,
 
 	return nil
 }
+
+func (r *ReconcileJaeger) getSelectorsForConfigMaps(instanceName string, components []string) labels.Selector {
+	nameReq, _ := labels.NewRequirement("app.kubernetes.io/name", selection.DoubleEquals, []string{util.Truncate(instanceName, 63)})
```
Comment (jpkrohling) [inline pkg/controller/jaeger/configmap.go:72]: Document why you are swallowing the error.

**Review by jpkrohling — COMMENTED** [review]
Looks good, just need to properly format the imports and document why errors are being swallowed.

File: pkg/controller/jaeger/configmap.go (hunk 2 above)
Comment (rubenvp8510) [reply pkg/controller/jaeger/configmap.go:73]: Done

File: pkg/controller/jaeger/configmap.go (hunk 1 above)
Comment (rubenvp8510) [reply pkg/controller/jaeger/configmap.go]: Ups.. my bad. Fixed

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**